src/bsp/src/stm32/src/SocketFactory.cpp

```diff
@@ -0,0 +1,30 @@
+#include "SocketFactory.h"
+#include "TCPClient.h"
+#include <lwip/sockets.h>
+#include <sys/types.h>
+
+std::optional<TCPClient> SocketFactory::createTCPClient(const char* address,
+                                                        uint32_t port,
+                                                        const ILogger& logger) {
+
+    // Creating the socket
+    int sockfd = lwip_socket(AF_INET, SOCK_STREAM, 0);
+    if (sockfd < 0) {
+        logger.log(LogLevel::Error, "Could not open socket");
+        return {};
+    }
+
+    // Creating the server address
+    struct sockaddr_in serverAddr {};
+    serverAddr.sin_family = AF_INET;
+    serverAddr.sin_port = htons(port);
+    serverAddr.sin_addr.s_addr = inet_addr(address);
+
+    // Connect the socket
+    if (lwip_connect(sockfd, (sockaddr*)&serverAddr, sizeof(serverAddr)) < 0) {
+        logger.log(LogLevel::Error, "Could not connect to server %s : %d", address, port);
```

Comment:
Is it an error in the posix build? Maybe we should put a warning since this may be a normal behavior if we want to connect to a phone (or any host really) via serial

Reply:
It was an error in posix. Changed both for info